src/pygeoprocessing/symbolic.py

```diff
@@ -72,30 +70,31 @@ def evaluate_raster_calculator_expression(
 
     # remove any raster bands that don't have corresponding symbols in the
     # expression
-    active_symbols = sorted(
-        [str(x) for x in sympy.parsing.sympy_parser.parse_expr(
-            expression).free_symbols])
-    raster_op = sympy.lambdify(active_symbols, expression, 'numpy')
-    raster_op_source = inspect.getsource(raster_op)
-    if not active_symbols:
-        raise ValueError(
-            'Symbolic expression reduces to a constant and does not need '
-            'evaluation. See inferred implementation:\n%s' % raster_op_source)
+    active_symbols = set()
+    for tree_node in ast.walk(ast.parse(expression)):
+        if isinstance(tree_node, ast.Name):
+            active_symbols.add(tree_node.id)
 
     LOGGER.debug(
-        'evaluating: %s\nactive symbols: %s\nraster_op:\n%s',
-        expression, active_symbols, raster_op_source)
+        'evaluating: %s\nactive symbols: %s\n',
+        expression, sorted(active_symbols))
     symbol_list, raster_path_band_list = zip(*[
         (symbol, raster_path_band) for symbol, raster_path_band in
         sorted(symbol_to_path_band_map.items()) if symbol in active_symbols])
 
+    missing_symbols = set(active_symbols) - set(symbol_list)
+    if len(missing_symbols) > 0:

```

Comment:
Can I nitpick here and request a Pythonic

```suggestion
    if missing_symbols:
```